### DISCUSSION TITLE
Problem when synchronizing (delete or add roles) the same user to 2 different teams

### DISCUSSION THREAD

**Discussion by alexv96** [spatie/laravel-permission#1921] [Q&A] [answered]
I have been registering several users testing and I have a problem that I do not know if it is only mine that when updating the roles of 1 user, I delete the roles of 1 team, ie.

Team A:
User cristinini@gmail.com has the roles of salesman and sales manager.

Team B:
User cristinini@gmail.com has the roles of viewer, salesman and sales manager. If I delete the salesman role, it deletes the roles of both TEAMS. 

I use this method:
`$user->syncRoles($request->get('roles'));
`
It only happens when I use this method, I would have to use another method provided by the library to avoid these problems or not?

I attach a graphical example
This image is how the roles are at the moment. It is only in 1 team
![image](https://user-images.githubusercontent.com/33232796/140857708-cefba5e4-cec5-4d80-bd5c-d8d11d516d19.png)

Now I added role in another team to the same user 3 and I deleted the roles and the previous team.
![image](https://user-images.githubusercontent.com/33232796/140857805-8e583e59-8d75-4459-9cf7-c6dc3212a8bd.png)

**Version**
"php": "^7.3|^8.0",
"laravel/framework": "^8.65",
"spatie/laravel-permission": "^5.1",
Mysql 5.1.1

**Comment by erikn69** [comment, marked as answer]
Always show your laravel, package and php versions, no one can guess
**Looks like that problem was fixed in the latest version #1888** 

If you use `$user->syncRoles($request->get('roles'));` and `$request->get('roles')` only has one role_id(2), it will detach all roles and put only role_id(2), it's the expected behaivor, read the documentation
https://github.com/spatie/laravel-permission/blob/193e50e44dce7b289f475a7df8782c9be0467e39/src/Traits/HasRoles.php#L196-L201

**Comment by erikn69** [comment]
```
Version
"php": "^7.3|^8.0",
"laravel/framework": "^8.65",
"spatie/laravel-permission": "^5.1",
Mysql 5.1.1
```
😄 that is your `composer.json`, it doesn't help to anyone, you are really new on this, use `composer show -i` for the installed versions, `composer update` for get the latest
